uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Validator/DateTest.php

```diff
@@ -34,7 +34,11 @@ public function testCanValidateDateInDifferentFormats($date, $format)
   public function datesDataProvider()
   {
     return [
-      [date('Y-m-d'), true],
+      [CRM_Utils_Date::processDate(date('Y-m-d')), true],
+      [CRM_Utils_Date::processDate(date('Y-m')), true],
+      [CRM_Utils_Date::processDate(date('Y')), true],
+      [CRM_Utils_Date::processDate(date('m')), true],
+      [date('Y-m-d'), false],
```

Comment:
CRM_Utils_Date::processDate() always returns YmdHis format so even if we pass only year it converts it to YmdHis (as we assume to get this format in BAO currently). Anyway I've added also additional date() values to check and only full date('YmdHis') is valid in our case.